lib/resources/azure/azure_generic_resource.rb

```diff
@@ -16,6 +16,8 @@ class AzureGenericResource < AzureResourceBase
     attr_accessor :filter, :total, :counts, :name, :type, :location, :probes
 
     def initialize(opts = {})
+      warn '[DEPRECATED] use specific azure resources instead. See https://github.com/inspec/inspec/issues/3131'
```

Comment:
Might want to mention that 'azure_generic_resource' was used - otherwise this might be a confusing message.